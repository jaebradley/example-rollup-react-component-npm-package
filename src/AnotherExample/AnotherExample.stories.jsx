import React from 'react';

import AnotherExample from './index.jsx';

export default {
  title: 'AnotherExample',
  component: AnotherExample,
};

const Template = (args) => <AnotherExample {...args} />;

export const AnotherBasicExample = Template.bind({});

export const ExampleWithCustomName = Template.bind({});
ExampleWithCustomName.args = { name: 'SomeOtherName' };

