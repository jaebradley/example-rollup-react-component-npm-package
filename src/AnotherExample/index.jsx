import React from 'react';
import PropTypes from 'prop-types';

import './AnotherExample.css';

const AnotherExample = ({ name }) => (
  <div className="another-example">
    This is another example by
    { name }
  </div>
);

AnotherExample.propTypes = {
  name: PropTypes.string,
};

AnotherExample.defaultProps = {
  name: 'jaebaebae',
};

export default AnotherExample;
