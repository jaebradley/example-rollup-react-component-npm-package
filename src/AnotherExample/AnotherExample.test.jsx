import React from 'react';
import {
  shallow,
} from 'enzyme';
import renderer from 'react-test-renderer';

import AnotherExample from '.';

describe('AnotherExample', () => {
  describe('when name prop is defined', () => {
    it('renders name prop', () => {
      const anExample = shallow(<AnotherExample name="Some Name" />);
      expect(anExample.text()).toEqual('This is another example bySome Name');
    });

    it('renders correctly', () => {
      const tree = renderer.create(<AnotherExample name="Some Name" />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('when name prop is not defined', () => {
    it('renders with default name prop', () => {
      const anExample = shallow(<AnotherExample />);
      expect(anExample.text()).toEqual('This is another example byjaebaebae');
    });

    it('renders correctly', () => {
      const tree = renderer.create(<AnotherExample />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
