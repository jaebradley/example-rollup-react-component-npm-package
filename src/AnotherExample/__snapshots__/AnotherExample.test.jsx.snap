// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnotherExample when name prop is defined renders correctly 1`] = `
<div
  className="another-example"
>
  This is another example by
  Some Name
</div>
`;

exports[`AnotherExample when name prop is not defined renders correctly 1`] = `
<div
  className="another-example"
>
  This is another example by
  jaebaebae
</div>
`;
